src/pages/wallet-import-keystore/wallet-import-keystore.spec.ts

```diff
@@ -300,4 +301,242 @@ describe("WalletImportKeystorePage", () => {
+  })
+
+  it("should return false when the name has zero length and true when the file imported name is atleast one character", () => {
+    comp.fileImportedName = "";
+    expect(comp.canShowFilePasswordInputAndImportedName()).toBe(false);
+
+    comp.fileImportedName = "more than one char it is!";
+    expect(comp.canShowFilePasswordInputAndImportedName()).toBe(true);
+  })
+
+  it("should return false when the password has zero length and the file import name can be shown and true when the file import password is atleast one character", () => {
+    spyOn(comp, "canShowFilePasswordInputAndImportedName").and.returnValue(true);
+
+    comp.filePassword = "";
+    expect(comp.canShowFileImportNameInput()).toBe(false);
+
+    comp.filePassword = "more than one char it is!";
+    expect(comp.canShowFileImportNameInput()).toBe(true);
+  })
+
+  it("should return false when the file import name cannot be shown and that name has zero characters and true when the file import name is shown and atleast one character", () => {
+    spyOn(comp, "canShowFileImportNameInput").and.returnValue(true);
+
+    comp.fileWalletNameImport = "";
+    expect(comp.canShowFileImportButton()).toBe(false);
+
+    comp.fileWalletNameImport = "more than one char it is!";
+    expect(comp.canShowFileImportButton()).toBe(true);
+  })
+
+  it("should call importWalletByCurrentKeystoreInfo if the keystore data is valid", () => {
+    spyOn(comp, "keystoreFileDataIsValid").and.returnValue(true);
+    spyOn(comp, "importWalletByCurrentKeystoreInfo");
+
+    comp.importByFile();
+
+    expect(comp.importWalletByCurrentKeystoreInfo).toHaveBeenCalled();
+  })
+
+  it("should set clipboard password to invalid when preparing the wallet with an invalid password", (done) => {
+    spyOn(comp, "prepareWallet").and.returnValue(null);
+    comp.importWalletByCurrentKeystoreInfo("clipboard").then(data => {
+      expect(comp.clipboardPasswordIsInvalid).toBe(true);
+      done();
+    });
+  })
+
+  it("should set file password to invalid when preparing the wallet with an invalid password", (done) => {
+    spyOn(comp, "prepareWallet").and.returnValue(null);
+    comp.importWalletByCurrentKeystoreInfo("file").then(data => {
+      expect(comp.filePasswordIsInvalid).toBe(true);
+      done();
+    });
+  })
+
+  it("should prepare the wallet correctly by file", () => {
+    spyOn(keyStoreService, "decryptKeyStore").and.returnValue("SOME_PRIVATE_KEY");
+    spyOn(walletService, "generateId").and.returnValue("WALLET_ID");
+
+    comp.fileWalletNameImport = "name";
+    comp.keyStore = <any>{};
+    comp.filePassword = "pass123";
+
+    let wallet = comp.prepareWallet("file");
+
+    expect(wallet).toEqual({
+      id: "WALLET_ID",
+      type: "local",
+      name: "name",
+      publicKey: null,
+      keyStore: comp.keyStore,
+      lastUpdateTime: null
+    });
+
+    expect(keyStoreService.decryptKeyStore).toHaveBeenCalledWith(comp.keyStore, "pass123");
```

Comment:
Just out of curiosity: why does the keystore need to be decrypted when preparing the wallet?